src/components/Filters/CompanyReceivedFilter.jsx

```diff
@@ -0,0 +1,205 @@
+    }
+    if ( dayjs( fromDate ).isBefore( minDate ) ) {
+      errs.push( "'From' date must be after " +
+          dayjs( minDate ).format( DATE_VALIDATION_FORMAT ) )
+    }
+    if ( dayjs( throughDate ).isAfter( maxDate ) ) {
+      errs.push( "'Through' date must be before " +
+          dayjs( maxDate ).format( DATE_VALIDATION_FORMAT ) )
+    }
+
+    return errs;
+  }, [ fromDate, throughDate ] )
+
+  const handleDateChange = () => {
+    let _throughDate = throughDate;
+    let _fromDate = fromDate
+    if ( _fromDate && !dayjs( fromDate ).isValid() ) {
+      fromRef.current.value = '';
+      _fromDate = ''
+    }
+    if ( _throughDate && !dayjs( throughDate ).isValid() ) {
+      throughRef.current.value = '';
+      _throughDate = ''
+    }
+    const isDateDifferent = dateFrom !== _fromDate ||
+        dateThrough !== _throughDate;
+    if ( isDateDifferent ) {
+      dispatch( changeDates( fieldName, _fromDate, _throughDate ) )
+    }
+  };
+
+  const inputFromClassName = useMemo( () => {
+    const style = [ 'a-text-input' ]
+    if ( dayjs( fromDate ).isBefore( minDate ) ||
+        dayjs( fromDate ).isAfter( throughDate ) ) {
+      style.push( 'a-text-input__error' )
+    }
+    return style.join( ' ' )
+  }, [ fromDate, throughDate ] );
+
+  const inputThroughClassName = useMemo( () => {
+    const style = [ 'a-text-input' ]
+    if ( dayjs( throughDate ).isAfter( maxDate ) ||
+        dayjs( throughDate ).isBefore( fromDate ) ) {
+      style.push( 'a-text-input__error' )
+    }
+    return style.join( ' ' )
+  }, [ fromDate, throughDate ] );
+
+  return <CollapsibleFilter title={title}
+                            className="aggregation date-filter">
+        <div>
+            <ul className="date-inputs">
+                <li>
+                    <label className="a-label a-label__heading body-copy"
+                           htmlFor={`${ fieldName }-from`}>
+                        From
+                    </label>
+                    <div className="m-btn-inside-input">
+                        <input id={`${ fieldName }-from`}
+                               className={inputFromClassName}
+                               onBlur={handleDateChange}
+                               onChange={evt => setFromDate( evt.target.value )}
+                               onKeyDown={handleKeyDownFromDate}
+                               min={minDate}
+                               max={maxDate}
+                               ref={fromRef}
+                               placeholder={DATE_VALIDATION_FORMAT}
+                               type="date"
+                               value={fromDate}
+                        />
+                        <button className="a-btn a-btn__link"
+                                onClick={() => handleClear( 'from' )}>
+                            {iconMap.getIcon( 'delete' )}
+                            <span className="u-visually-hidden">Clear</span>
```

Comment:
Say what is being cleared. This would be read out loud in an accessible context.